shopinvader_api_address_shipping_note/schemas.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.addons.shopinvader_api_address.schemas import (
+    DeliveryAddressCreate,
+    DeliveryAddressUpdate,
+)
+from odoo.addons.shopinvader_schema_address.schemas import DeliveryAddress
+
+
+class ShippingAddressNote(DeliveryAddress, extends=True):
+    """
+    Shipping Address
+    """
+
+    shipping_note: str | None = None
+
+    @classmethod
+    def from_res_partner(cls, odoo_rec):
+        res = super().from_res_partner(odoo_rec)
+        res.shipping_note = odoo_rec.shipping_note or None
+
+        return res
+
+
+class ShippingAddressNoteCreate(DeliveryAddressCreate, extends=True):
+    """
+    Creation of Shipping Address
+    """
+
+    shipping_note: str | None = None
+
+    def to_res_partner_vals(self) -> dict:
+        vals = super().to_res_partner_vals()
+
+        vals["shipping_note"] = self.shipping_note
+
+        return vals
+
+
+class ShippingAddressNoteUpdate(DeliveryAddressUpdate, extends=True):
+    """
+    Update of Shipping Address
+    """
+
+    shipping_note: str | None = None
+
+    def to_res_partner_vals(self) -> dict:
+        vals = super().to_res_partner_vals()
+        shipping_note = self.shipping_note
```

Comment:
With partial update, it's not required to set the field into the json you pass to odoo when updating the address. With your code, you' don't make the difference between to cases:
* The field is not set into the json -> value is None in python and we don't want to update it  
* The field is set to null into the json -> value is None in python and we want to update it...

To covers both cases you must call the `model_dump` to retrieve fields really given to instanciate the model:

```python
values = self.model_dump(exclude_unset=True, include=["shipping_note"])
if "shipping_note" in values:
    vals["shipping_note"] = values["shipping_note"
return vals]
```

Reply:
Thanks, I updated the code